authz.go

```diff
@@ -40,15 +40,14 @@ func PermissionAllowed(p Permission, ps []Permission) bool {
 	}
 
 	for _, perm := range ps {
-		permID := ID(0)
-		if perm.ID != nil {
-			permID = *perm.ID
-			if !permID.Valid() {
-				return false
+		if perm.ID == nil {
+			if perm.Action == p.Action && perm.Resource == p.Resource && perm.OrgID == p.OrgID {
+				return true
+			}
+		} else {
+			if (*perm.ID).Valid() && *perm.ID == pID && perm.Action == p.Action && perm.Resource == p.Resource && perm.OrgID == p.OrgID {
```

Comment:
`(*perm.ID).Valid()`

You shouldn't need to check this and `*perm.ID == pID`; we already checked that `pID` is valid on line 37, so anything equal to that must also be valid.

Reply:
Good point 👍